pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -4238,9 +4238,76 @@ func (c *Cloud) getInstancesByIDs(instanceIDs []*string) (map[string]*ec2.Instan
 }
 
 func (c *Cloud) getInstancesByNodeNames(nodeNames []string, states ...string) ([]*ec2.Instance, error) {
+	ec2Instances := []*ec2.Instance{}
 	names := aws.StringSlice(nodeNames)
+
+	instanceIDs := make([]*string, 0, len(nodeNames))
+	namesWithoutIDs := make([]*string, 0, len(nodeNames))
+	for _, name := range names {
+		instanceID, err := c.nodeNameToProviderID(types.NodeName(*name))
+		if err != nil {
+			namesWithoutIDs = append(namesWithoutIDs, name)
+		} else {
+			instanceIDString := string(instanceID)
+			instanceIDs = append(instanceIDs, &instanceIDString)
+		}
+	}
+
+	if len(namesWithoutIDs) > 0 {
+		instancesFromNames, err := c.getInstancesByPrivateDNSNames(namesWithoutIDs, states...)
+		if err != nil {
+			return nil, err
+		}
+		ec2Instances = append(ec2Instances, instancesFromNames...)
+	}
+
+	if len(instanceIDs) > 0 {
+		instancesFromIDs, err := c.getInstancesByInstanceIDs(instanceIDs, states...)
+		if err != nil {
+			return nil, err
+		}
+		ec2Instances = append(ec2Instances, instancesFromIDs...)
+	}
+
+	if len(ec2Instances) == 0 {
+		glog.V(3).Infof("Failed to find any instances %v", nodeNames)
+		return nil, nil
+	}
+	return ec2Instances, nil
+}
+
+func (c *Cloud) getInstancesByInstanceIDs(instanceIDs []*string, states ...string) ([]*ec2.Instance, error) {
 	ec2Instances := []*ec2.Instance{}
+	for i := 0; i < len(instanceIDs); i += filterNodeLimit {
+		end := i + filterNodeLimit
```

Comment:
I thik we need another constant (than filterNodeLimit) when retrieving by id because we can 1000+ instances with this method

Reply:
Agreed, I need to look into how big we can make that.

Reply:
There is no official documentation but from informal discussion with AWS there does not seem to be a limit and 1000 is sure to work. Maybe use 1000 for a start?

Reply:
Done!